File: 167. Two Sum II - Input array is sorted.cpp
class Solution {

public:

    vector<int> twoSum(vector<int>& numbers, int target) {

        int i(0), j(numbers.size()-1);

        vector<int> result;

        while(i<j){

            if(numbers[i] + numbers[j] == target){

                result.push_back(i+1);

                result.push_back(j+1);

                return result;

            }

            else if (numbers[i] + numbers[j] < target)

                i++;

            else if (numbers[i] + numbers[j] > target)

                j--;

        }

        result.push_back(-1);

        result.push_back(-1);

        return result;

    }

};

File: 167. Two Sum II - Input array is sorted.py
class Solution(object):

    def twoSum(self, numbers, target):

        """

        :type numbers: List[int]

        :type target: int

        :rtype: List[int]

        """

        i, j = 0, len(numbers)-1

        while i<j:

            if numbers[i] + numbers[j] == target:

                return [i+1, j+1]

            elif numbers[i] + numbers[j] < target:

                i = i + 1;

            else:

                j = j - 1;

        return [-1, -1]

            
